CMakeLists.txt

```diff
@@ -92,6 +92,14 @@ set(Trilinos_PREFIX "" CACHE STRING "Trilinos installation directory")
 option(SKIP_SIMMETRIX_VERSION_CHECK "enable at your own risk; it may result in undefined behavior" OFF)
 option(ENABLE_SIMMETRIX "Build with Simmetrix support" OFF)
 message(STATUS "ENABLE_SIMMETRIX: ${ENABLE_SIMMETRIX}")
+option(ENABLE_EGADS "Build with EGADS support" OFF)
+message(STATUS "ENABLE_EGADS: ${ENABLE_EGADS}")
+if(ENABLE_EGADS)
+  add_definitions(-DHAVE_EGADS)
```

Comment:
To add compiler definitions you should be [`target_compile_definitions`](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html). Since you use this option in the header it will need to be public. Because this leaks into the user's downstream code you should use a prefixed name. I.e. `PUMI_HAVE_EGADS`

Reply:
Thanks for taking another look at this @jacobmerson. I wrote the `-DHAVE_EGADS` definition to be consistent with the simmetrix definition:  
https://github.com/SCOREC/core/blob/30a332d97469e9719313e312512b2d407e912f54/CMakeLists.txt#L97-L99
Based on the way that `HAVE_SIMMETRIX` is used, it seems convenient to use `add_definitions` so that it applies to all the executables so that multiple calls to `target_compile_definitions` can be skipped, but I am aware that this approach is considered bad practice in modern cmake. In any case I thought it would be best to stay consistent, though I'm not tied to the current approach.

Reply:
I would prefer that all of those other `HAVE_XXX` definitions get prefixed eventually and you have a valid point about consistency. The problem with adding compile definitions without a namespace is that any other consuming project linking against `pumi_egads` has that compile definition on their target. So, what if a consumer wants to use PUMI compiled with egads but they have conditional code behind `HAVE_EGADS` that they don't want to compile for some reason...currently they cannot do that since compile definitions are infectious. I think @cwsmith has last word on this sort of thing since he has to deal with all the xsdk people/rules.

The issue with the non-target version of `add_definitions` (especially in the top level cmake file) is that it will be applied to every single target and library at the current or lower sub-directory level. So, even random libraries that have nothing to do with egads are getting that compile definition added. Since these compile definitions will be infectious to anyone who uses any SCOREC target it has the potential to cause issues for down stream users.

Reply:
@cwsmith do you have any thoughts on the concerns raised by @jacobmerson?

Reply:
@jacobmerson is correct.  We should have all the `*HAVE*`,`*ENABLE*`, etc... variables prefixed with `PUMI_` to avoid conflicts with packages that use PUMI.  +1 for using `target_compile_definitions` as well.

I created an issue to clean this up in `develop` after this is merged. https://github.com/SCOREC/core/issues/337